### PR TITLE
Fix to Issue 698 and other misc changes

### PR DESCRIPTION
**Pull request by manojampalam**
Fix to https://github.com/PowerShell/Win32-OpenSSH/issues/698 - Create job object, add ssh-shellhost and ensure all its child process and tied to its lifetime.

Other changes include changes to debug* statements in posix adapter that may cause recursion/stack overflow issues.

### CODE REVIEW

File: contrib/win32/win32compat/termio.c [hunk 1]
```diff
@@ -107,7 +107,8 @@ ReadThread(_In_ LPVOID lpParameter)
 
 			if (!ReadFile(WINHANDLE(pio), pio->read_details.buf,
 				pio->read_details.buf_size, &read_status.transferred, NULL)) { 
-				read_status.error = GetLastError();				
+				debug4("ReadThread - ReadFile failed %d, io:%p", GetLastError(), pio);
```
Comment (bingbing8) [inline contrib/win32/win32compat/termio.c]: minor suggest: error:%d

File: contrib/win32/win32compat/termio.c [hunk 2]
```diff
@@ -126,7 +127,8 @@ ReadThread(_In_ LPVOID lpParameter)
 	} else {
 		if (!ReadFile(WINHANDLE(pio), pio->read_details.buf,
 		    pio->read_details.buf_size, &read_status.transferred, NULL)) {
-			read_status.error = GetLastError();			
+			debug4("ReadThread - ReadFile failed %d, io:%p", GetLastError(), pio);
```
Comment (bingbing8) [inline contrib/win32/win32compat/termio.c]: minor suggest: error:%d

File: contrib/win32/win32compat/termio.c [hunk 3]
```diff
@@ -209,7 +211,7 @@ WriteThread(_In_ LPVOID lpParameter)
 		if (!WriteFile(WINHANDLE(pio), pio->write_details.buf, write_status.to_transfer,
 		    &write_status.transferred, NULL)) {
 			write_status.error = GetLastError();
-			debug("WriteThread - ReadFile WriteFile %d, io:%p", GetLastError(), pio);
+			debug4("WriteThread - ReadFile WriteFile %d, io:%p", GetLastError(), pio);
```
Comment (bingbing8) [inline contrib/win32/win32compat/termio.c]: Remove ReadFile from the mesage. add error:%d

File: contrib/win32/win32compat/shell-host.c [hunk 4]
```diff
@@ -1595,6 +1597,21 @@ wmain(int ac, wchar_t **av)
...
+		return -1;
+	}
+
+	memset(&job_info, 0, sizeof(JOBOBJECT_EXTENDED_LIMIT_INFORMATION));
+	job_info.BasicLimitInformation.LimitFlags = JOB_OBJECT_LIMIT_KILL_ON_JOB_CLOSE;
+
+	if (!SetInformationJobObject(job, JobObjectExtendedLimitInformation, &job_info, sizeof(job_info)) ||
+		!AssignProcessToJobObject(job, GetCurrentProcess())) {
+		printf_s("cannot associate job object: %d", GetLastError());
+		return -1;
```
Comment (bingbing8) [inline contrib/win32/win32compat/shell-host.c:1612]: I assume the handle is closed when process exits.

**Review by bingbing8 — APPROVED** [review]

File: contrib/win32/win32compat/shell-host.c (hunk 4 above)
Comment (manojampalam) [reply contrib/win32/win32compat/shell-host.c:1612]: Right.

**Pull request merged**